client/state/user-settings/actions.js

```diff
@@ -116,22 +137,40 @@ export const removeUnsavedUserSetting = ( settingName ) => ( {
 export function setUserSetting( settingName, value ) {
 	return ( dispatch, getState ) => {
 		const settings = getUserSettings( getState() );
-
-		if ( get( settings, settingName ) === undefined ) {
+		const settingKey = Array.isArray( settingName ) ? settingName : settingName.split( '.' );
```

Comment:
It feels like this logic could be abstracted and reused where necessary. 

Some tests could be useful for this action, too.

Reply:
I extracted `setUserSetting` into the `thunks` directory and improved tests in https://github.com/Automattic/wp-calypso/pull/49760/commits/4e13faeb55669696e76d8682042ad72740218fb5